te3/tutorials/data-security/data-security-testing.md

```diff
@@ -0,0 +1,81 @@
+---
+uid: data-security-testing
+title: Testing RLS/OLS
+author: Kurt Buhler
+updated: 2023-03-02
+applies_to:
+  editions:
+    - edition: Desktop
+      none: x
+    - edition: Business
+    - edition: Enterprise
+---
+# Testing Data Security with Impersonation
+
+
+![Data Security Visual Abstract](~/images/data-security/data-security-testing-visual-abstract.png)
+
+
+---
+
+__DAX Queries__, __Pivot Grid__, or __Preview Data__ enable testing of Data Security in Tabular Editor. It is recommended to *always* test Data Security with any changes to configuration, to mitigate risk of incorrect RLS/OLS implementation and the consequences thereof.
+
+> [!IMPORTANT]
+> Testing Data Security with Impersonation using Tabular Editor 3 is limited to dataset hosted in the Power BI Datasets service. TE3 Desktop Licenses cannot benefit from this feature. This is because roles are assigned in the Power BI Service.
+
+---
+
+
+- [__About Data Security and RLS/OLS:__](data-security-about.md) A functional overview of <span style="color:#01a99d">RLS</span> & <span style="color:#8d7bae">OLS</span>.
+- [__Modify/Setup an RLS Configuration:__](data-security-setup-rls.md) How to configure <span style="color:#01a99d">RLS</span> in a dataset.
+- [__Modify/Setup an OLS Configuration:__](data-security-setup-ols.md) How to configure <span style="color:#8d7bae">OLS</span> in a dataset.
+- __Testing RLS/OLS with Impersonation (This Article):__ How to easily validate Data Security with Tabular Editor.
+
+---
+
+## Testing with Impersonation
+
+__Data Security can be easily tested using _Impersonation_ in Tabular Editor 3.__ Impersonation is a feature that lets you view a query result as a model Role or User. It is similar to the _'View As Role...'_ feature in the Power BI service, with two key differences:
+
+1. The User requires __dataset Build permissions__ in addition to Role assignment & Dataset Read access. 
```

Comment:
The user being impersonated; since the user doing the impersonation is making changes to the data model they have write access which supersedes build access. Is it even possible for someone with build permissions only to connect to a Power BI dataset with Tabular Editor, actually?

Reply:
I would specify that, as it is ambiguous at the moment.

Reply:
Done!